file.go

```diff
@@ -779,3 +779,18 @@ func putPageHeader(h *format.PageHeader) {
 	*h = format.PageHeader{}
 	pageHeaderPool.Put(h)
 }
+
+func (f *File) readAt(p []byte, off int64) (int, error) {
+	return readAt(f.reader, p, off)
+}
+
+func readAt(r io.ReaderAt, p []byte, off int64) (n int, err error) {
+	n, err = r.ReadAt(p, off)
+	if n == len(p) {
```

Comment:
Maybe we should be more conservative and only ignore `io.EOF` instead of dropping any error. It would also be nice to log the error as a warning, but I don't think we have a logger available in the `File` API at the moment.

Reply:
Just  wondering, what other error you had in mind that might be relevant to return ?

Reply:
I can't really think of anything specific, but `io.Reader` is an interface and the underlying implementation that can return any error, even a user-defined one. I just thought it might be useful to propagate it to the caller instead of dropping it, but it's probably an edge case is unlikely to happen.

Reply:
Right, I see. We can leave edge cases that we aren't aware of for now.